src/meta/src/hummock/compaction/level_selector.rs

```diff
@@ -626,4 +627,175 @@ pub mod tests {
         let compaction = selector.pick_compaction(2, &levels, &mut levels_handlers);
         assert!(compaction.is_none());
     }
+
+    #[test]
+    fn test_manual_compaction_picker_l0() {
+        let config = Arc::new(CompactionConfigBuilder::new().max_level(4).build());
+        let selector = DynamicLevelSelector::new(config, Arc::new(RangeOverlapStrategy::default()));
+        let l0 = generate_l0_with_overlap(vec![
+            generate_table(0, 1, 0, 500, 1),
+            generate_table(1, 1, 0, 500, 1),
+        ]);
+        assert_eq!(l0.sub_levels.len(), 2);
+        let levels = vec![
+            generate_level(1, vec![]),
+            generate_level(2, vec![]),
+            generate_level(3, vec![]),
+            Level {
+                level_idx: 4,
+                level_type: LevelType::Nonoverlapping as i32,
+                table_infos: vec![
+                    generate_table(2, 1, 0, 100, 1),
+                    generate_table(3, 1, 101, 200, 1),
+                    generate_table(4, 1, 222, 300, 1),
+                ],
+                total_file_size: 0,
+                sub_level_id: 0,
+            },
+        ];
+        assert_eq!(levels.len(), 4);
+        let levels = Levels {
+            levels,
+            l0: Some(l0),
+        };
+        let mut levels_handler = (0..5).into_iter().map(LevelHandler::new).collect_vec();
+
+        // pick_l0_to_sub_level
+        {
+            let option = ManualCompactionOption {
+                sst_ids: vec![0, 1],
+                key_range: KeyRange {
+                    left: vec![],
+                    right: vec![],
+                    inf: true,
+                },
+                internal_table_id: HashSet::default(),
+                level: 0,
+            };
+            let task = selector
+                .manual_pick_compaction(1, &levels, &mut levels_handler, option)
+                .unwrap();
+            assert_eq!(task.input.input_levels.len(), 2);
+            assert_eq!(task.input.input_levels[0].level_idx, 0);
+            assert_eq!(task.input.input_levels[1].level_idx, 0);
+            assert_eq!(task.input.target_level, 0);
+        }
```

Comment:
I think we can check  pending_task status after pick_compaction. 

For expected, the files that picked by compaction should be added to pending_task in level_handler

Reply:
Fixed.